cfssl_scan.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/cloudflare/cfssl/scan"
+)
+
+var scanUsageText = `cfssl scan -- scan a host for issues
+
+Usage of scan:
+        cfssl scan [-short] [SCANFAMILY] HOST
+        cfssl scan [-short] [-scanner regexp] HOST
+        cfssl scan [-list] [SCANFAMILY]*
+
+Arguments:
+        HOST:    Host to scan (including port)
+        SCANFAMILY:	 Regexp selecting family of related scanners to run
+Flags:
+`
+
+var scanFlags = []string{"short", "list", "scanner"}
+
+func scanMain(args []string) (err error) {
+	var scanFamilyRegexp, scannerRegexp *regexp.Regexp
+	var host string
+	scan.Short = Config.short
+	if Config.list {
+		if len(args) == 0 {
+			for _, scanFamily := range scan.AllFamilies {
+				fmt.Printf("%s:\n\t%s\n\n", scanFamily.Name, scanFamily.Description)
+			}
+		}
+		for len(args) > 0 {
+			var scanFamilyName string
+			scanFamilyName, args, err = popFirstArgument(args)
+			if err != nil {
+				return
+			}
+			scanFamilyRegexp = regexp.MustCompile(scanFamilyName)
+			for _, scanFamily := range scan.AllFamilies {
+				var isMatch bool
+				isMatch, err = regexp.MatchString(scanFamilyName, scanFamily.Name)
+				if err != nil {
+					return
+				}
+				if isMatch {
+					fmt.Printf("%s:\n\t%s\n", scanFamily.Name, scanFamily.Description)
+					for _, scanner := range scanFamily.Scanners {
+						fmt.Printf("\t%s:\n\t\t%s", scanner.Name, scanner.Description)
+					}
+				}
+			}
+		}
+	} else {
+		if Config.scanner != "" {
+			scannerRegexp = regexp.MustCompile(Config.scanner)
+		}
+		for len(args) > 1 {
```

Comment:
This logic needs some work. calling cfssl scan with no arguments causes scanFamily.Name to be nil and 
			matches := scanFamilyRegexp.MatchString(scanFamily.Name)
to cause a panic.